share/lib/reports.awk

```diff
@@ -465,7 +465,7 @@ END {
   printf "%s%-80.80s%s\n", white, padding, normal;
 
   # Set return code
-  if (run_error+audit_error+audit_noncompliant+enforce_error != 0) {
+  if (run_error+audit_error+enforce_error != 0) {
```

Comment:
i validated, but this means that we want non compliant to be ok ?

Reply:
I'm not sur I understand either. Actually, I'm not sure of the semantic of the return code, but `non compliance` is as much an error as `impossible to enforce`, I think. If my ssh key does't match the expected set of authorized one, it's as important as if in enforce, it was that and I wasn't able to change it. Or is this audit error? And if so, what is `noncompliant` ?

Reply:
It means we consider non-compliant to be normal and expected agent behavior. Errors or aborted runs are not normal and should be fixed by the user.

Reply:
In fact I think error reports should not change the exit code to 1. Aborted agents are the only real problems the agent itself can have.

Reply:
I agree with that, but if so: why audit error and enforce error change it to 1?

Reply:
(ie, we diferentiate from "expected" errors that are just part of the business domain, and "unexpect" defaicts that should be processed outside of business domain. An enforce in error is expected, an agent run crash is not)

Reply:
They shouldn't :), il make another PR.

Reply:
https://github.com/Normation/rudder-agent/pull/299 and https://github.com/Normation/rudder-agent/pull/298 for the doc